spec.bs

```diff
@@ -243,8 +250,200 @@ Each {{URLPattern}} object has an associated <dfn for=URLPattern>hash component<
   1. Return [=this=]'s [=URLPattern/hash component=]'s [=component/pattern string=].
 </div>
 
+<div algorithm>
+  The <dfn method for="URLPattern">test(|input|, |baseURL|)</dfn> method steps are:
+
+  1. Let |result| be the result of [=match=] given [=this=], |input|, and |baseURL|.
+  1. If |result| is null, return false.
+  1. Return true.
+</div>
+
+<div algorithm>
+  The <dfn method for="URLPattern">exec(|input|, |baseURL|)</dfn> method steps are:
+
+  1. Return the result of [=match=] given [=this=], |input|, and |baseURL|.
+</div>
+
+<div algorithm>
+  To perform a <dfn>match</dfn> given a {{URLPattern}} |urlpattern|, a {{URLPatternInput}} |input|, and an optional string |baseURLString|:
+
+  1. Let |protocol| be the empty string.
+  1. Let |username| be the empty string.
+  1. Let |password| be the empty string.
+  1. Let |hostname| be the empty string.
+  1. Let |port| be the empty string.
+  1. Let |pathname| be the empty string.
+  1. Let |search| be the empty string.
+  1. Let |hash| be the empty string.
+  1. Let |inputs| be an empty array.
```

Comment:
In spec-land we use [lists](https://infra.spec.whatwg.org/#lists) so as not to be confused with developer-modifiable JavaScript arrays. So you'd use `an empty [=list=]` and `[=list/Append=]` instead of `Push` below.